stmt.go

```diff
@@ -335,13 +336,15 @@ func (st *statement) ExecContext(ctx context.Context, args []driver.NamedValue)
 	if err := ctx.Err(); err != nil {
 		return nil, err
 	}
+
 	Log := ctxGetLog(ctx)
 
 	st.Lock()
 	defer st.Unlock()
 	if st.conn == nil {
 		return nil, driver.ErrBadConn
 	}
+	st.ctx = ctx
```

Comment:
Clear st.ctx in st.closeNotLocked.

Reply:
reset context to context.Background . ok?

Reply:
A nil would be better.